### DISCUSSION TITLE
resvg PNG output filesize

### DISCUSSION THREAD

**Discussion by david-buck** [vercel/satori#245] [Ideas]
I'm in the process of migrating my OG image layouts from my customised solution based on [vercel/ogimage](https://github.com/vercel/og-image), and things are generally pretty smooth. 

One thing I've noticed that that the generated files are very different in file size, with the Satori-based ones being about 2x for comparable images. 

Is there a chance that something like oxipng could be incorporated into the process?

**Comment by shuding** [comment]
It’s configurable, but also a trade off, that setting a higher compression rate results slower generation speed. I’ll see what we can make a better balance here.

**Comment by david-buck** [comment]
That would be great. 

Anecdotally, the images generated by resvg are more accomodating of a compression pass than the ones generated by puppeteer: 

<img width="664" alt="image" src="https://user-images.githubusercontent.com/5119820/197078658-2750150e-b347-442f-bfa6-0d36cf879733.png">

This example is running using oxipng on fastest setting on almost identical images.

Resvg:
in 82,919 bytes  -> out 13,568 bytes
83.6% saving

Puppeteer:
in 45,950 bytes -> out 33,838 bytes
26.4% saving

**Comment by styfle** [comment]
Here's a related feature request for jpeg: https://github.com/vercel/satori/issues/235
